pipelines/apDetectorVisitQualityCore.yaml

```diff
@@ -12,3 +12,10 @@ tasks:
       atools.simpleSky: SimpleDiaPlot
       python: |
         from lsst.analysis.tools.atools import *
+  analyzeTrailedDiaSrcCore:
```

Comment:
Interstingly, I know the answer only because of the normal dia source count failing. In the following code, `analyzeTrailedDiaSrcCore` fails because there is no fakes (not sure why it is insisting there be faces), but `analyzeTrailedDiaSrcCore` succeeds because it finds the correct dataset. So if its not present that one part of the analysis fails but the others continue on. It shouldn't be much of a problem, I guess? I'm not sure how we want this to work if there isn't anything there, is a failure to run message enough or should it be smart enough to skip?

```(lsst-scipipe-7.0.1) [smart@sdfrome002 analysis_tools]$ pipetask run -p pipelines/apDetectorVisitQualityCore.yaml -b /sdf/group/rubin/u/smart/code/workspaces/trailedSourceTables_analysis_tools_test/repo/butler.yaml -i ap_verify-output -o trailedTest -d "skymap='hsc_rings_v1' and instrument='HSC'" --register-dataset-types --instrument lsst.obs.subaru.HyperSuprimeCam
lsst.pipe.base.quantum_graph_builder INFO: Processing pipeline subgraph 1 of 2 with 1 task(s).
lsst.pipe.base.quantum_graph_builder INFO: Iterating over query results to associate quanta with datasets.
lsst.pipe.base.quantum_graph_builder INFO: Initial bipartite graph has 10 quanta, 42 dataset nodes, and 40 edges from 10 query row(s).
lsst.pipe.base.quantum_graph_builder INFO: Processing pipeline subgraph 2 of 2 with 1 task(s).
py.warnings WARNING: /sdf/group/rubin/sw/conda/envs/lsst-scipipe-7.0.1/share/eups/Linux64/pipe_base/gfacbe7f57e+607ede7504/python/lsst/pipe/base/all_dimensions_quantum_graph_builder.py:500: FutureWarning: Dataset type(s) ['fakes_goodSeeingDiff_assocDiaSrc'] are not registered; this will be an error after v26.
  with builder.butler.registry.queryDataIds(**result.query_args).materialize() as common_data_ids:

lsst.pipe.base.quantum_graph_builder INFO: Iterating over query results to associate quanta with datasets.
lsst.pipe.base.quantum_graph_builder CRITICAL: Initial data ID query returned no rows, so QuantumGraph will be empty.
lsst.pipe.base.quantum_graph_builder CRITICAL: Dataset type fakes_goodSeeingDiff_assocDiaSrc is not registered.
lsst.pipe.base.quantum_graph_builder CRITICAL: To reproduce this query for debugging purposes, run Registry.queryDataIds with these arguments:
lsst.pipe.base.quantum_graph_builder CRITICAL:   dimensions=['band', 'instrument', 'detector', 'physical_filter', 'visit'],
lsst.pipe.base.quantum_graph_builder CRITICAL:   dataId={},
lsst.pipe.base.quantum_graph_builder CRITICAL:   where="skymap='hsc_rings_v1' and instrument='HSC'",
lsst.pipe.base.quantum_graph_builder CRITICAL:   datasets=['fakes_goodSeeingDiff_assocDiaSrc'],
lsst.pipe.base.quantum_graph_builder CRITICAL:   collections=['trailedTest'],
lsst.pipe.base.quantum_graph_builder INFO: Dropping task analyzeAssocDiaSrcCore because no quanta remain.
lsst.pipe.base.quantum_graph_builder INFO: Generated 10 quanta for task analyzeTrailedDiaSrcCore.
lsst.ctrl.mpexec.cmdLineFwk INFO: QuantumGraph contains 10 quanta for 1 task, graph ID: '1699050511.0522678-4059886'
Quanta          Tasks          
------ ------------------------
    10 analyzeTrailedDiaSrcCore
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Preparing execution of quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 58, visit: 59150, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Constructing task and executing quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 58, visit: 59150, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Execution of task 'analyzeTrailedDiaSrcCore' on quantum {instrument: 'HSC', detector: 58, visit: 59150, ...} took 0.115 seconds
lsst.ctrl.mpexec.mpGraphExecutor INFO: Executed 1 quanta successfully, 0 failed and 9 remain out of total 10 quanta.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Preparing execution of quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 4, visit: 59134, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Constructing task and executing quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 4, visit: 59134, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Execution of task 'analyzeTrailedDiaSrcCore' on quantum {instrument: 'HSC', detector: 4, visit: 59134, ...} took 0.105 seconds
lsst.ctrl.mpexec.mpGraphExecutor INFO: Executed 2 quanta successfully, 0 failed and 8 remain out of total 10 quanta.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Preparing execution of quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 51, visit: 59160, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Constructing task and executing quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 51, visit: 59160, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Execution of task 'analyzeTrailedDiaSrcCore' on quantum {instrument: 'HSC', detector: 51, visit: 59160, ...} took 0.077 seconds
lsst.ctrl.mpexec.mpGraphExecutor INFO: Executed 3 quanta successfully, 0 failed and 7 remain out of total 10 quanta.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Preparing execution of quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 5, visit: 59142, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Constructing task and executing quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 5, visit: 59142, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Execution of task 'analyzeTrailedDiaSrcCore' on quantum {instrument: 'HSC', detector: 5, visit: 59142, ...} took 0.086 seconds
lsst.ctrl.mpexec.mpGraphExecutor INFO: Executed 4 quanta successfully, 0 failed and 6 remain out of total 10 quanta.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Preparing execution of quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 0, visit: 59134, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Constructing task and executing quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 0, visit: 59134, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Execution of task 'analyzeTrailedDiaSrcCore' on quantum {instrument: 'HSC', detector: 0, visit: 59134, ...} took 0.085 seconds
lsst.ctrl.mpexec.mpGraphExecutor INFO: Executed 5 quanta successfully, 0 failed and 5 remain out of total 10 quanta.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Preparing execution of quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 43, visit: 59160, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Constructing task and executing quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 43, visit: 59160, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Execution of task 'analyzeTrailedDiaSrcCore' on quantum {instrument: 'HSC', detector: 43, visit: 59160, ...} took 0.053 seconds
lsst.ctrl.mpexec.mpGraphExecutor INFO: Executed 6 quanta successfully, 0 failed and 4 remain out of total 10 quanta.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Preparing execution of quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 0, visit: 59142, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Constructing task and executing quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 0, visit: 59142, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Execution of task 'analyzeTrailedDiaSrcCore' on quantum {instrument: 'HSC', detector: 0, visit: 59142, ...} took 0.083 seconds
lsst.ctrl.mpexec.mpGraphExecutor INFO: Executed 7 quanta successfully, 0 failed and 3 remain out of total 10 quanta.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Preparing execution of quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 50, visit: 59150, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Constructing task and executing quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 50, visit: 59150, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Execution of task 'analyzeTrailedDiaSrcCore' on quantum {instrument: 'HSC', detector: 50, visit: 59150, ...} took 0.097 seconds
lsst.ctrl.mpexec.mpGraphExecutor INFO: Executed 8 quanta successfully, 0 failed and 2 remain out of total 10 quanta.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Preparing execution of quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 11, visit: 59142, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Constructing task and executing quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 11, visit: 59142, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Execution of task 'analyzeTrailedDiaSrcCore' on quantum {instrument: 'HSC', detector: 11, visit: 59142, ...} took 0.082 seconds
lsst.ctrl.mpexec.mpGraphExecutor INFO: Executed 9 quanta successfully, 0 failed and 1 remain out of total 10 quanta.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Preparing execution of quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 5, visit: 59134, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Constructing task and executing quantum for label=analyzeTrailedDiaSrcCore dataId={instrument: 'HSC', detector: 5, visit: 59134, ...}.
lsst.ctrl.mpexec.singleQuantumExecutor INFO: Execution of task 'analyzeTrailedDiaSrcCore' on quantum {instrument: 'HSC', detector: 5, visit: 59134, ...} took 0.093 seconds
lsst.ctrl.mpexec.mpGraphExecutor INFO: Executed 10 quanta successfully, 0 failed and 0 remain out of total 10 quanta.
```

Reply:
Oh OK, if it just says "dropping Task X because no quanta remain" as above and still does the other metrics without falling over, that seems reasonable.

I think an older version of `analyzeAssocDiaSrcCore` had hardwired `fakesType` and `coaddName` parameters and it was since updated to have defaults that can be overridden.